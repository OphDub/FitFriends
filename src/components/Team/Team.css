.Team {
  padding: 10px; }
