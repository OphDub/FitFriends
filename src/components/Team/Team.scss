.Team {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: lightgrey;
  min-height: 75%;
  padding: 15px;
  margin: 0 10px 10px 10px;
  border-radius: 10px;
}

.Team-header {
  margin: 10px 0;
}