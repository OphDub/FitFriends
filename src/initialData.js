export const mockTeam = [
  {
    name: 'Felipe',
    workoutTally: 6,
    workoutsPosted: 2,
  },
  {
    name: 'Fion',
    workoutTally: 5,
    workoutsPosted: 3,
  },
  {
    name: 'Morgan',
    workoutTally: 3,
    workoutsPosted: 4,
  },
]

export const mockWorkoutHistory = [
  {
    workoutName: 'The Grind',
    workoutDesc: 'Opposite movement workout with some running at the end!',
    exercises: [
      {reps: '10', exercise: 'Burpees'},
      {reps: '8-10', exercise: 'Overhead Press(Barbell)'},
      {reps: '8-10', exercise: 'Standing Bent Row(Barbell)'},
      {reps: '8-10', exercise: 'Standing Kettlebell Swings'},
    ],
  }
]

export const mockUserProfile = {
  userImage: 'http://www.ixgetfit.com/wp-content/uploads/2017/10/Brooke-Wells-350x350-300x300.jpg',
  userName: 'Brooke'
}

export const mockWorkouts = [
  {
    workoutName: 'local sample workout name 1',
    workoutDesc: 'local sample workout desc',
    exercises: [
      { reps: 10, exercise: 'sample exercise1'},
      { reps: 10, exercise: 'sample exercise2'},
    ]
  },
  {
    workoutName: 'local sample workout name 2',
    workoutDesc: 'local sample workout desc',
    exercises: [
      { reps: 10, exercise: 'sample exercise1'},
      { reps: 10, exercise: 'sample exercise2'},
    ]
  },
]