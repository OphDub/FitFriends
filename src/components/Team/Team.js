import React from 'react';
import './Team.css';
import { Teammate }from '../Teammate/Teammate';

export const Team = ({team}) => {
    const renderedTeam = team.map(teammate =>
    <Teammate {...teammate}/>
  )

  return(
    <section className="Team">
      {renderedTeam}
    </section>
  )
}