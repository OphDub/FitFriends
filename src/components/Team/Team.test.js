/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { Team } from './Team';
import { mockTeam } from '../../initialData';

describe('TEAM', () => {
  it('should match snapshot', () => {
    const renderedComponent = shallow(<Team team={mockTeam}/>, { disableLifecycleMethods: true });

    expect(renderedComponent).toMatchSnapshot();
  });
});