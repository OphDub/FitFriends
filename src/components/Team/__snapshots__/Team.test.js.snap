// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEAM should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    team={
        Array [
            Object {
              "UID": "randomKey1",
              "name": "Felipe",
              "workoutTally": 6,
              "workoutsPosted": 2,
            },
            Object {
              "UID": "randomKey2",
              "name": "Fion",
              "workoutTally": 5,
              "workoutsPosted": 3,
            },
            Object {
              "UID": "randomKey3",
              "name": "Morgan",
              "workoutTally": 3,
              "workoutsPosted": 4,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="Team-header"
>
          Team Members
</h3>,
        Array [
          <Unknown
            UID="randomKey1"
            name="Felipe"
            workoutTally={6}
            workoutsPosted={2}
/>,
          <Unknown
            UID="randomKey2"
            name="Fion"
            workoutTally={5}
            workoutsPosted={3}
/>,
          <Unknown
            UID="randomKey3"
            name="Morgan"
            workoutTally={3}
            workoutsPosted={4}
/>,
        ],
      ],
      "className": "Team",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Team Members",
          "className": "Team-header",
        },
        "ref": null,
        "rendered": "Team Members",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": "randomKey1",
        "nodeType": "function",
        "props": Object {
          "UID": "randomKey1",
          "name": "Felipe",
          "workoutTally": 6,
          "workoutsPosted": 2,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "randomKey2",
        "nodeType": "function",
        "props": Object {
          "UID": "randomKey2",
          "name": "Fion",
          "workoutTally": 5,
          "workoutsPosted": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "randomKey3",
        "nodeType": "function",
        "props": Object {
          "UID": "randomKey3",
          "name": "Morgan",
          "workoutTally": 3,
          "workoutsPosted": 4,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="Team-header"
>
            Team Members
</h3>,
          Array [
            <Unknown
              UID="randomKey1"
              name="Felipe"
              workoutTally={6}
              workoutsPosted={2}
/>,
            <Unknown
              UID="randomKey2"
              name="Fion"
              workoutTally={5}
              workoutsPosted={3}
/>,
            <Unknown
              UID="randomKey3"
              name="Morgan"
              workoutTally={3}
              workoutsPosted={4}
/>,
          ],
        ],
        "className": "Team",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Team Members",
            "className": "Team-header",
          },
          "ref": null,
          "rendered": "Team Members",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": "randomKey1",
          "nodeType": "function",
          "props": Object {
            "UID": "randomKey1",
            "name": "Felipe",
            "workoutTally": 6,
            "workoutsPosted": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "randomKey2",
          "nodeType": "function",
          "props": Object {
            "UID": "randomKey2",
            "name": "Fion",
            "workoutTally": 5,
            "workoutsPosted": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "randomKey3",
          "nodeType": "function",
          "props": Object {
            "UID": "randomKey3",
            "name": "Morgan",
            "workoutTally": 3,
            "workoutsPosted": 4,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
